File: src/main/java/pl/yalgrin/playnite/simplesync/web/FeatureResource.java
package pl.yalgrin.playnite.simplesync.web;

import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;
import pl.yalgrin.playnite.simplesync.dto.FeatureDTO;
import pl.yalgrin.playnite.simplesync.helper.SingleExecutorHelper;
import pl.yalgrin.playnite.simplesync.service.FeatureService;
import reactor.core.publisher.Mono;

@RestController
@RequestMapping("/api/feature")
@RequiredArgsConstructor
public class FeatureResource {
    private final FeatureService service;
    private final SingleExecutorHelper singleExecutorHelper;

    @GetMapping("/{id}")
    public Mono<FeatureDTO> getFeature(@PathVariable Long id) {
        return service.findById(id);
    }

    @PostMapping("/save")
    public Mono<FeatureDTO> saveFeature(@RequestBody FeatureDTO dto, @RequestParam String clientId) {
        return singleExecutorHelper.runOnExecutor(service.saveObject(dto, clientId));
    }

    @PostMapping("/delete")
    public Mono<Void> deleteFeature(@RequestBody FeatureDTO dto, @RequestParam String clientId) {
        return singleExecutorHelper.runOnExecutor(service.deleteObject(dto, clientId));
    }
}

File: src/main/kotlin/pl/yalgrin/playnite/simplesync/web/library/FeatureResource.kt
package pl.yalgrin.playnite.simplesync.web.library

import org.springframework.web.bind.annotation.*
import pl.yalgrin.playnite.simplesync.dto.objects.FeatureDTO
import pl.yalgrin.playnite.simplesync.helper.SingleExecutorHelper
import pl.yalgrin.playnite.simplesync.service.objects.FeatureService
import reactor.core.publisher.Mono

@RestController
@RequestMapping("/api/feature")
class FeatureResource(
    private val service: FeatureService,
    private val singleExecutorHelper: SingleExecutorHelper
) {

    @GetMapping("/{id}")
    fun getFeature(@PathVariable id: Long): Mono<FeatureDTO> {
        return service.findById(id)
    }

    @PostMapping("/save")
    fun saveFeature(@RequestBody dto: FeatureDTO): Mono<FeatureDTO> {
        return singleExecutorHelper.runOnExecutor(service.saveObject(dto))
    }

    @PostMapping("/delete")
    fun deleteFeature(@RequestBody dto: FeatureDTO): Mono<Void> {
        return singleExecutorHelper.runOnExecutor(service.deleteObject(dto))
    }
}